src/ethereum/cancun/vm/instructions/system.py

```diff
@@ -522,15 +523,18 @@ def selfdestruct(evm: Evm) -> None:
     # beneficiary).
     set_account_balance(evm.env.state, originator, U256(0))
```

Comment:
Balance transfer also has to only happen under certain conditions i.e. if the contract was created in the same transaction or if the beneficiary and originator are different.